cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -63,26 +64,23 @@ public SecureStoreHandler(SecureStore secureStore, SecureStoreManager secureStor
   public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
                      @PathParam("key-name") String name) throws BadRequestException, IOException {
     if (!EntityId.isValidStoreKey(name)) {
-      throw new BadRequestException("Improperly formatted name. The name can contain lower case alphabets, " +
-                                    "numbers, _, and -");
+      throw new BadRequestException("Improperly formatted secure key name. The name can contain lower case " +
+                                      "alphabets, " + "numbers, _, and -");
     }
     SecureKeyCreateRequest secureKeyCreateRequest = parseBody(httpRequest, SecureKeyCreateRequest.class);
 
     if (secureKeyCreateRequest == null) {
-      throw new BadRequestException("Unable to parse the request. The request body should be of the following format" +
-                                      "{\n" +
-                                      "  \"description\" :  \"<description>\"\n" +
-                                      "  \"data\"        :  \"<data>\"  //utf-8\n" +
-                                      "  \"properties\"  :  {\n" +
-                                      "    \"key\"  :  \"value\"\n" +
-                                      "    ...\n" +
-                                      "  }");
+      SecureKeyCreateRequest dummy = new SecureKeyCreateRequest("<description>", "<data>",
+                                                                ImmutableMap.of("key", "value"));
+      throw new BadRequestException("Unable to parse the request. The request body should be of the following format." +
+                                      " \n" + dummy);
```

Comment:
use Gson with pretty printing turned on, because this won't display a json, right?